ingestion_tools/scripts/importers/annotation.py

```diff
@@ -24,28 +35,199 @@ class AnnotationObject(TypedDict):
 class AnnotationSource(TypedDict):
     columns: str
     shape: str
-    glob_string: str
     file_format: str
+    delimiter: str | None
     binning: int | None
     order: str | None
     filter_value: str | None
+    is_visualization_default: bool | None
+    mask_label: str | None
 
 
 class AnnotationMap(TypedDict):
     metadata: dict[str, Any]
     sources: list[AnnotationSource]
 
 
-class BaseAnnotationSource:
+# This class is basically a global var that lets us cache metadata and identifiers for annotations,
+# so we can generate non-conflicting sequential identifiers for annotations as they're imported.
+class AnnotationIdentifierHelper:
+    next_identifier: dict[str, int] = defaultdict(partial(int, 100))
+    cached_identifiers: dict[str, int] = {}
+    loaded_vs_metadatas: dict[str, bool] = {}
```

Comment:
Could this be a set?

Reply:
Nope, it needs to be indexed by a specific set of annotation metadata, since it's basically a gross global variable that knows how to calculate identifiers for all annotations imported by a particular run of the import script :'( it's not a great design, but basically we need to share state between multiple different AnnotationImporter instances somehow.